Empty POSTs to /labels when uploading.

This lovely error has been seen a few times in the last few days, causing a few alarms too.

```
2015-07-03 09:38:53,822 - [ERROR] - from play in play-akka.actor.default-dispatcher-55788
Cannot invoke the action, eventually got an error: com.amazonaws.AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 4E39J2LRLHJFVCPULHBPNI0IEJVV4KQNSO5AEMVJF66Q9ASUAAJG)

2015-07-03 09:38:53,823 - [ERROR] - from application in play-akka.actor.default-dispatcher-55788


! @6mk69n2cn - Internal server error, for (POST) [/metadata/524e7ba22ba5eaa06cf81380dbd73bd60a77c88b/labels] ->

play.api.Application$$anon$1: Execution exception[[AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 4E39J2LRLHJFVCPULHBPNI0IEJVV4KQNSO5AEMVJF66Q9ASUAAJG)]]
	at play.api.Application$class.handleError(Application.scala:296) ~[app.jar:0.1]
	at play.api.DefaultApplication.handleError(Application.scala:402) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at scala.Option.map(Option.scala:146) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [app.jar:0.1]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40) [app.jar:0.1]
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248) [app.jar:0.1]
	at scala.concurrent.Promise$class.complete(Promise.scala:55) [app.jar:0.1]
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.processBatch$1(BatchingExecutor.scala:67) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:82) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [app.jar:0.1]
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58) [app.jar:0.1]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [app.jar:0.1]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:401) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [app.jar:0.1]
Caused by: com.amazonaws.AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 4E39J2LRLHJFVCPULHBPNI0IEJVV4KQNSO5AEMVJF66Q9ASUAAJG)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1078) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:726) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:461) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:296) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:3139) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.updateItem(AmazonDynamoDBClient.java:1334) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.doUpdateItem(UpdateItemImpl.java:104) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.updateItem(UpdateItemImpl.java:86) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.Table.updateItem(Table.java:212) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.DynamoDB$$anonfun$update$1.apply(DynamoDB.scala:153) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.DynamoDB$$anonfun$update$1.apply(DynamoDB.scala:145) ~[app.jar:0.1]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[app.jar:0.1]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[app.jar:0.1]
	... 6 common frames omitted

2015-07-03 09:38:55,077 - [INFO] - from request in play-akka.actor.default-dispatcher-55768
77.91.250.202 - "POST /metadata/524e7ba22ba5eaa06cf81380dbd73bd60a77c88b/labels HTTP/1.1" ERROR "https://media.gutools.co.uk/upload" 21ms

2015-07-03 09:38:55,078 - [ERROR] - from play in play-akka.actor.default-dispatcher-55768
Cannot invoke the action, eventually got an error: com.amazonaws.AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: UTF21K5A27UH7N41L5BRUSRA4RVV4KQNSO5AEMVJF66Q9ASUAAJG)

2015-07-03 09:38:55,078 - [ERROR] - from application in play-akka.actor.default-dispatcher-55768


! @6mk69n495 - Internal server error, for (POST) [/metadata/524e7ba22ba5eaa06cf81380dbd73bd60a77c88b/labels] ->

play.api.Application$$anon$1: Execution exception[[AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: UTF21K5A27UH7N41L5BRUSRA4RVV4KQNSO5AEMVJF66Q9ASUAAJG)]]
	at play.api.Application$class.handleError(Application.scala:296) ~[app.jar:0.1]
	at play.api.DefaultApplication.handleError(Application.scala:402) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at scala.Option.map(Option.scala:146) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320) [app.jar:0.1]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [app.jar:0.1]
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40) [app.jar:0.1]
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248) [app.jar:0.1]
	at scala.concurrent.Promise$class.complete(Promise.scala:55) [app.jar:0.1]
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [app.jar:0.1]
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) [app.jar:0.1]
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.processBatch$1(BatchingExecutor.scala:67) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:82) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch$$anonfun$run$1.apply(BatchingExecutor.scala:59) [app.jar:0.1]
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72) [app.jar:0.1]
	at akka.dispatch.BatchingExecutor$Batch.run(BatchingExecutor.scala:58) [app.jar:0.1]
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:41) [app.jar:0.1]
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:401) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [app.jar:0.1]
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [app.jar:0.1]
Caused by: com.amazonaws.AmazonServiceException: ExpressionAttributeValues contains invalid value: One or more parameter values were invalid: An string set  may not be empty for key :value (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: UTF21K5A27UH7N41L5BRUSRA4RVV4KQNSO5AEMVJF66Q9ASUAAJG)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1078) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:726) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:461) ~[app.jar:0.1]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:296) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:3139) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.updateItem(AmazonDynamoDBClient.java:1334) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.doUpdateItem(UpdateItemImpl.java:104) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.updateItem(UpdateItemImpl.java:86) ~[app.jar:0.1]
	at com.amazonaws.services.dynamodbv2.document.Table.updateItem(Table.java:212) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.DynamoDB$$anonfun$update$1.apply(DynamoDB.scala:153) ~[app.jar:0.1]
	at com.gu.mediaservice.lib.aws.DynamoDB$$anonfun$update$1.apply(DynamoDB.scala:145) ~[app.jar:0.1]
	at scala.concurrent.impl.Fu
```

--- a/metadata-editor/app/controllers/EditsController.scala
+++ b/metadata-editor/app/controllers/EditsController.scala
@@ -244,7 +244,7 @@ object EditsController extends Controller with ArgoHelpers {
   )
 
   val listForm: Form[List[String]] = Form(
-     single[List[String]]("data" -> list(text))
+     single[List[String]]("data" -> list(nonEmptyText))
   )
 
 }

